packages/markers/src/browser/problem/problem-widget.ts

```diff
@@ -37,6 +39,24 @@ export class ProblemWidget extends TreeWidget {
         this.addClipboardListener(this.node, 'copy', e => this.handleCopy(e));
     }
 
+    protected deflateForStorage(node: ITreeNode): object {
+        const result = super.deflateForStorage(node) as any;
+        if (UriSelection.is(node) && node.uri) {
+            result.uri = node.uri.toString();
```

Comment:
```
  export function is(e: any): e is UriSelection {
        return e && e["uri"] instanceof URI
    }
```

Is there really a need to check for && node.uri ? seems like UriSelection.is would do.

Reply:
Also I don't understand what Selection is in that context.. seems like is should be like URINode...